sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -426,6 +432,36 @@ public ReadWriteSpan updateName(String name) {
     return this;
   }
 
+  @Override
+  public Span addLink(SpanContext spanContext, Attributes attributes) {
+    if (spanContext == null || !spanContext.isValid()) {
+      return this;
+    }
+    if (attributes == null) {
+      attributes = Attributes.empty();
+    }
+    synchronized (lock) {
+      if (hasEnded) {
+        logger.log(Level.FINE, "Calling addLink() on an ended Span.");
+        return this;
+      }
+      if (links == null) {
+        links = new ArrayList<>(spanLimits.getMaxNumberOfLinks());
+      }
+      if (links.size() < spanLimits.getMaxNumberOfLinks()) {
+        links.add(
+            LinkData.create(
```

Comment:
the tiniest nit...you could create this outside of the synchronized block. Would trade off some allocations for a tiny bit less time under the lock.